nibabel/nicom/dicomwrappers.py

```diff
@@ -528,12 +540,15 @@ def image_shape(self):
         # Store frame indices
         self._frame_indices = frame_indices
         if n_dim < 4:  # 3D volume
-            return rows, cols, n_frames
+            return rows, cols, self._nframes
         # More than 3 dimensions
         ns_unique = [len(np.unique(row)) for row in self._frame_indices.T]
+        if len(ns_unique) == 3:
+            # derived volume is included
+            ns_unique.pop(1)
```

Comment:
I don't really follow this guard. Can you elaborate?

Reply:
I'm not sure this addresses the problem properly / gracefully, but I'm unsure how to proceed. I've come across a derived image included in the diffusion DICOM, which is currently not properly handled. I thought this meant 5D data wasn't yet handled, but in fact there is a test specifically for this introduced in 3f04e3c60acd7d28204d0c81d4294cb6e7397cfe (maybe @matthew-brett can share some wisdom)
https://github.com/nipy/nibabel/blob/ad0b13f24a7b91a7be9ce70a46bb79901b4b5f48/nibabel/nicom/tests/test_dicomwrappers.py#L656-L682